datadog/tests/resource_datadog_user_test.go

```diff
@@ -218,6 +218,49 @@ func TestAccDatadogUser_UpdateRole(t *testing.T) {
 	})
 }
 
+func TestAccDatadogUser_ReEnableRoleUpdate(t *testing.T) {
+	t.Parallel()
+	ctx, accProviders := testAccProviders(context.Background(), t)
+	username := strings.ToLower(uniqueEntityName(ctx, t)) + "@example.com"
+	accProvider := testAccProvider(t, accProviders)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: accProviders,
+		CheckDestroy:      testAccCheckDatadogUserV2Destroy(accProvider),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckDatadogUserConfigRoleUpdate1(username),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDatadogUserV2Exists(accProvider, "datadog_user.foo"),
+					resource.TestCheckResourceAttr("datadog_user.foo", "email", username),
+					resource.TestCheckResourceAttr("datadog_user.foo", "name", "Test User"),
+					resource.TestCheckResourceAttr("datadog_user.foo", "verified", "false"),
+					resource.TestCheckResourceAttr("datadog_user.foo", "roles.#", "2"),
+					testCheckUserHasRole("datadog_user.foo", "data.datadog_role.st_role"),
+					testCheckUserHasRole("datadog_user.foo", "data.datadog_role.ro_role"),
+				),
+			},
+			{
+				// Destroy the user resource by passing data source resource only
+				Config: roleDatasources,
```

Comment:
can we add a check that the user does not exist for sure (or is disabled)?

Reply:
Added a call to ListUsers endpoint to see if there is a user disabled with the handle